File: civitasRuby/lib/sorpresa_pagar_cobrar.rb
# encoding:utf-8
module Civitas

  class SorpresaPagarCobrar < Sorpresa
  
    def initialize(texto, valor)
      super(texto)
      @valor = valor
    end
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        super(actual, todos)
        todos.at(actual).modificar_saldo(@valor)
      end
    end
    
    def to_string
      return super.to_string + "\nValor: #{@valor}"
    end
    
  public_class_method :new  
  end
end

File: civitas/src/civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class SorpresaPagarCobrar extends Sorpresa {
    
    private int valor;

    public SorpresaPagarCobrar(int valor, String texto) {
        super(texto);
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
    
    @Override
    public String toString() {
        String salida = "Valor: " + valor + super.toString();
        return salida;
    }
    
}
